compiler-core/src/analyse.rs

```diff
@@ -1343,3 +1343,72 @@ fn get_compatible_record_fields<A>(
 
     compatible
 }
+
+/// Given a type, return a list of all the types it depends on
+fn get_type_dependencies(type_name: SmolStr, typ: &TypeAst) -> Vec<SmolStr> {
```

Comment:
Best take as a reference here as it is not always consumed
```suggestion
fn get_type_dependencies(type_name: &SmolStr, typ: &TypeAst) -> Vec<SmolStr> {
```